### PR TITLE
Add test for SelectHowToPay.

### PR DESCRIPTION
**Pull request by kberg**
Made it more sophsiticated so it's easier to read and more clicks can be verified.

### CODE REVIEW

File: src/components/PaymentWidgetMixin.ts [hunk 1]
```diff
@@ -3,13 +3,14 @@ import {CardName} from '../CardName';
 import {CardModel} from '../models/CardModel';
 import {ResourceType} from '../ResourceType';
 
+type ResourcesWithRates = 'titanium' | 'steel' | 'microbes' |'floaters';
```
Comment (bafolts) [inline src/components/PaymentWidgetMixin.ts:6]: We can eventually improve upon having to provide these types manually but for the time being doing this is great.

File: src/components/PaymentWidgetMixin.ts [hunk 2]
```diff
@@ -58,23 +59,26 @@ export const PaymentWidgetMixin = {
...
-              (this as any)['titanium'] * this.getResourceRate('titanium') -
-              (this as any)['steel'] * this.getResourceRate('steel') -
-              (this as any)['microbes'] * this.getResourceRate('microbes') -
-              (this as any)['floaters'] * this.getResourceRate('floaters') -
-              ((this as any)['science'] ?? 0);
+      const ta = this as any;
+      const heatMc = ta['heat'] ?? 0;
+      const titaniumMc = (ta['titanium'] ?? 0) * this.getResourceRate('titanium');
+      const steelMc = (ta['steel'] ?? 0) * this.getResourceRate('steel');
+      const microbesMc = (ta['microbes'] ?? 0) * this.getResourceRate('steel');
```
Comment (bafolts) [inline src/components/PaymentWidgetMixin.ts:66]: This should read `const microbesMc = (ta['microbes'] ?? 0) * this.getResourceRate('microbes');`

File: src/components/PaymentWidgetMixin.ts [hunk 3]
```diff
@@ -58,23 +59,26 @@ export const PaymentWidgetMixin = {
...
     },
-    setMaxValue: function(target: string, max?: number): void {
+    setMaxValue: function(target: ResourcesWithRates | 'science', max?: number): void {
       let currentValue: number = (this as any)[target];
       const cardCost: number = (this as any).$data.cost;
       let amountHave: number = max ?? (this as any).player[target];
 
       let amountNeed: number = cardCost;
-      if (['titanium', 'steel', 'microbes', 'floaters'].includes(target)) {
+      if (!(target === 'science')) {
```
Comment (bafolts) [inline src/components/PaymentWidgetMixin.ts:81]: The `target` will also be `heat`. When it is `heat` this block used to return `false` and will now return `true`.

File: src/components/SelectHowToPay.ts [hunk 4]
```diff
@@ -169,7 +170,7 @@ export const SelectHowToPay = Vue.component('select-how-to-pay', {
       }
 
       const requiredAmt = this.playerinput.amount || 0;
-      const totalSpentAmt = htp.heat + htp.megaCredits + (htp.steel * this.player.steelValue) + (htp.titanium * this.player.titaniumValue) + (htp.microbes * 2) + (htp.floaters * 3);
+      const totalSpentAmt = htp.heat + htp.megaCredits + (htp.steel * this.player.steelValue) + (htp.titanium * this.player.titaniumValue);
```
Comment (bafolts) [inline src/components/SelectHowToPay.ts:173]: Great change. Checks out.

File: src/components/common/Button.vue [hunk 5]
```diff
@@ -74,6 +74,10 @@ export default Vue.extend({
       // type
       if (this.type === 'max') {
         classes.push('btn-max'); // #TODO max_button?
+      } else if (this.type === 'plus') {
+        classes.push('btn-plus');
```
Comment (bafolts) [inline src/components/common/Button.vue:78]: Does this `btn-plus` classname exist in CSS? I can't find it.

**Review by bafolts — CHANGES_REQUESTED** [review]
Love seeing a component test. Want to learn more about btn-plus and btn-minus. Reading more I think they are related to the component test. Spotted at least one issue in `PaymentWidgetMixin` will need to address before merging.

File: src/components/PaymentWidgetMixin.ts (hunk 1 above)
Comment (kberg) [reply src/components/PaymentWidgetMixin.ts:6]: Thanks.

File: src/components/PaymentWidgetMixin.ts (hunk 2 above)
Comment (kberg) [reply src/components/PaymentWidgetMixin.ts:66]: Eep!

**Comment by kberg** [comment]
> Love seeing a component test. Want to learn more about btn-plus and btn-minus. Reading more I think they are related to the component test. Spotted at least one issue in `PaymentWidgetMixin` will need to address before merging.

btn-minus and btn-plus are added so it's easier for the test to find those buttons and press them.

**Review by bafolts — APPROVED** [review]

**Pull request merged**